README.md

```diff
@@ -14,8 +10,279 @@ But we need have a clear way to discover how you've been invited to a group with
 We also need to consider how to ensure our group data is replicated _enough_ - e.g. if a group is just me and my sister, then are there gonna be enough copies of the group data to be resilent/ timely.
 
 
---- 
+## Principles
+
+1. **group membership should be opaque**
+   - you shouldn't be able to guess who is in a group using public info
+     - shards people are using
+     - groupId (e.g. %a2148bs2K3klmhnaALsdaN90=.cloaked)
+2. **we support sympathetic replication**
+
+
+
+## Spec
+
+This work builds on the [ssb-meta-feeds-spec] (v1).
+
+We define two types of feeds that each peer will have:
+1. <div class='group-invite'></div> the group-invite feed
+2. <div class='group'></div>        group feeds
+
+
+```mermaid
+graph TB
+
+root(root)
+v1(v1)
+5(5) & b(b) & f(f)
+group-invite(group-invite):::groupInvite
+denmark(denmark):::group
+helsinki(helsinki):::group
+wellington(wellington):::group
+
+root --> v1 --> 5 --> helsinki
+         v1 --> b --> group-invite
+         v1 --> f --> denmark
+                f --> wellington
+
+classDef default stroke:none;
+classDef groupInvite fill: #BF2669, stroke:none, color:white;
+classDef group fill: #702A8C, stroke: none, color:white;
+```
+_Diagram showing an example layout of group-related feeds. Note that the shards in your
+use-case will likely not be those shown, see how they are determined below._
+
+
+### the group-invite feed
+
+This feed holds messages which help peers join groups (e.g. `group/registration`, `group/add-member` messages).
+This feed MUST be unique for each peer (a singleton).
+Each peer MUST replicate this feed from each of their peers (OR each peer they might possibly enter a group with).
```

Comment:
```suggestion
Each peer A who replicates the root metafeed of another B SHOULD also replicate B's invitation feed. 
```

I'm not sure if I understood what you were trying to say, so I wrote it like this instead. Let me know what you think